### PR TITLE
Move a few exit plugins to postbuild task

### PR DESCRIPTION
**Pull request by tkdchen**
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates
- [n/a] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/tasks/binary.py [hunk 1]
```diff
@@ -73,6 +73,9 @@ class BinaryPostBuildTask(plugin_based.PluginBasedTask):
             {"name": "compare_components"},
             {"name": "group_manifests"},
             {"name": "generate_maven_metadata"},
+            {"name": "verify_media", "required": False},
+            {"name": "push_floating_tags"},
+            {"name": "koji_tag_build"},
```
Comment (chmeliik) [inline atomic_reactor/tasks/binary.py:78]: koji_tag_build cannot run before koji_import

**Review by chmeliik — COMMENTED** [review]
The base classes for the plugins also need to be updated, as well as the PluginsRunners used in unit tests

And it might be good idea to remove the logic that checks if the build process failed, as postbuild plugins they probably don't need that anymore

**Comment by chmeliik** [comment]
Looks good, but maybe we should leave out the [Run koji_tag_build plugin in postbuild task](https://github.com/containerbuildsystem/atomic-reactor/pull/1824/commits/3330ac143a16e9eebeb9cea6364a5d5658fe77a2) commit for now and do it later together with koji_import

**Comment by tkdchen** [comment]
I've made part of changes to koji_import. At this moment, I think the plan can change a little :), let's keep this pr open and I'll add more commits related to koji_import here.

File: atomic_reactor/plugins/exit_release_build_reservation.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if build_info is None:
+            self.log.warning(
+                "Cannot get the reserved build %s from Brew/Koji.", reserved_build_id
+            )
+            return
+
+        if build_info["state"] != koji.BUILD_STATES["BUILDING"]:
+            self.log.debug("Reserved build %s is in state %s already. Skip cancelation.",
+                           reserved_build_id, build_info["state"])
+            return
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply atomic_reactor/plugins/exit_release_build_reservation.py:45]: After `CGImport` call, reserved build will be in COMPLETE state. It is actually intentional not to check `self.workflow.build_process_failed`. In my mind, this plugin can just care about the reservation only. Whatever the build process succeeds or not, the study of this plugin is to ensure the reserved build is released.

Can the word "release" confuse? Would it be clearer by changing it to "cancel"?

[the remark this answers is not in the document]
Comment (tkdchen) [reply atomic_reactor/plugins/exit_release_build_reservation.py:45]: Maybe this plugin should also:

```python
if not self.workflow.build_process_failed and build_info["state"] == koji.BUILD_STATES["BUILDING"]:
    raise some error
```

When it happens, it means the koji_import plugin is not set properly for a workable container build pipeline.

[the remark this answers is not in the document]
Comment (chmeliik) [reply atomic_reactor/plugins/exit_release_build_reservation.py:45]: +1, that makes sense to me

> Can the word "release" confuse? Would it be clearer by changing it to "cancel"?

+1 to this as well :)

[the remark this answers is not in the document]
Comment (tkdchen) [reply atomic_reactor/plugins/exit_release_build_reservation.py]: Updated.

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Review by hjmodi — APPROVED** [review]

**Pull request merged**